Sources/KalturaPlayer.swift

```diff
@@ -382,4 +392,33 @@ public enum KalturaPlayerError: PKError {
     @objc public func startBuffering() {
         pkPlayer.startBuffering()
     }
+    
+    internal func updateMediaEntryWithLoadedInterceptors(_ mediaEntry: PKMediaEntry, callback: @escaping (_ error: Error?) -> Void) {
```

Comment:
Correct, this was a mistake on our side.

Reply:
While adding this, my thoughts was maybe it is possible to have such errors returning from interceptor.apply(on: entry) that it is not possible to use this entry for playback (in some use cases) and we have to change Plugin config and try to do all the process one more time starting from kalturaOTTPlayer.loadMedia(options: mediaOptions).

At this moment I do not know if this will be applicable in real cases. But having a error callback will be nice to have and maybe use it in the future. Currently, yes it is true we are ignoring errors and it seems the we do not need such callback.

Reply:
For now, please remove the error - from interceptor protocol as well as this function.

Reply:
> For now, please remove the error - from interceptor protocol as well as this function.

Agree that needs to remove Error from `internal func updateMediaEntryWithLoadedInterceptors(_ mediaEntry: PKMediaEntry)`

But in the protocol `PKMediaEntryInterceptor ` it is useful to have error calback. `PKMediaEntry.sources` is optional array. 
In [iOS implementation](https://github.com/chausov/playkit-ios-broadpeak-smartlib/blob/107d09a53e8ac0ddd0216f458b93603783ad4437/PlayKitBroadpeak/Classes/BroadpeakMediaEntryInterceptor.swift#L82) I'm checking if we have sources -> ok lets do SmartLib flow, if not -> return error that this media entry can not me modified because there is no sources, etc. All errors that caused in SmartLib flow will be sent to messageBus.

In [Android implementation](https://github.com/kaltura/playkit-android-broadpeak-smartlib/blob/f02f487a6aecff721f0f5952f4019d678903d32d/broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java#L103) it is little differently implemented. But previously we agreed that protocol interface should be similar on both, with parameters: entry and error callback .

Reply:
Also error callback can be used for such cases as timeouts errors if the interceptor implementation requires many web calls, etc.

Reply:
If the input is bad (no sources), just log the error and don't touch the input. The player will gain nothing by looking at the error.
@x-NR-x @OrenMe WDYT?

Reply:
I agree, error is in the domain of the plugin which the player knows nothing about.
The interface should be strict - get `PKMediaEntry` and return `PKMediaEntry`.
Error handling of the plugin should be managed by the plugin.

We discussed this previously if to add protection measurements like handling thrown errors and long running plugins but it is an overhead and should be managed by the plugin as core can't decide use case for why a plugin will need to consume more or less time.

For fatal errors if plugin decides that he couldn't retrieve critical data to allow playback he can raise a critical error on the player message bus and it will be listened to by app that has the responsibility to stop the session and display an error screen.